### DISCUSSION TITLE
Writing network client, V examples

### DISCUSSION THREAD

**Discussion by eptx** [vlang/v#16195] [General]
I'm writing a server client in V, the Scala client code uses...

```
  val socket = new Socket
  socket.connect(new InetSocketAddress(host, port), port)
  val in = new BufferedInputStream(socket.getInputStream)
  val out = socket.getOutputStream
...    
val os = new ByteArrayOutputStream
```
What are the corresponding V libraries to use and are there examples? Please advise if there's a better place to post these questions.

**Comment by JalonSolov** [comment]
Depends on what you want to do.  There are both https://modules.vlang.io/vweb.html and https://github.com/nedpals/vex, as well as lower-level code like you show... you can check files in the `examples` subdir for some simple examples.

**Comment by Casper64** [comment]
If you want more examples with socket programming, for me it was really helpful to look at the source code and the code in the `_test.v` files in https://github.com/vlang/v/tree/master/vlib/net
